src/contracts/types/factories/GenSale_factory.ts

```diff
@@ -0,0 +1,944 @@
+/* Autogenerated file. Do not edit manually. */
```

Comment:
I know we've historically kept these generated type files in source control, but we shouldn't. It even states on the npm package page that we shouldn't and instead we can rely on the `postinstall` for the package.json

Reply:
Im not sure I understand the alternative here. How does that work when running the app locally or getting the sdk to be able to execute the sdk methods?

Is it worth doing that now? or in the genericized contract where our deadline isnt as tight

Reply:
It's true that we need these types to be able to write the methods that interact with the contract, but they are generated by the ABI which is in source control and should be kept there. These types can be generated from that using the `yarn typechain` script defined in the package.json, and so if you just added a single line for the postinstall to call that script and do it, it would eliminate the need to keep anything in the `contracts/types` directory at all.

Reply:
Also, did you manually copy this from the contract repo? I see it only uses 1 underscore in the naming which is off because it uses 2 in the files it generates. Run `yarn typechain` instead.

Reply:
Yes, I got it from the contract repo, I didnt realize It could generate it here. Ive updated the factory to be the version generated by the local typechain command